Name conflict when using actix-web

Hi!
I noticed a naming collision when using actix-web and SQLx together. Basically, if I have an endpoint `password`, the various routing macros will create a struct with name `password`. In Rust, a structure name cannot be used as an identifier for a variable, which is exactly what happens when using the `query!` macro with for example 'password' as a parameter. I don't exactly know how the macro works, so I sadly cannot give any suggestions on how to fix this.

Here's an example of the error I am getting when compiling:
```rust
error[E0308]: mismatched types
  --> backend/src/auth.rs:11:8
   |
11 |       if sqlx::query!(
   |  ________^
12 | |         r#"
13 | |         SELECT * FROM users WHERE
14 | |             uname = $1 AND  
15 | |            password = $2;
16 | |    "#,
17 | |        &payload.identifier,
18 | |         &payload.password,
19 | |     )
   | |     ^
   | |     |
   | |     expected struct `String`, found struct `password`
   | |     this expression has type `String`
   | |_____`password` is interpreted as a unit struct, not a new binding
   |       help: introduce a new binding instead: `other_password`
...
63 |   #[post("/password")]
   |   -------------------- unit struct defined here
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> backend/src/auth.rs:11:8
   |
11 |       if sqlx::query!(
   |  ________^
12 | |         r#"
13 | |         SELECT * FROM users WHERE
14 | |             uname = $1 AND  
15 | |            password = $2;
16 | |    "#,
17 | |        &payload.identifier,
18 | |         &payload.password,
19 | |     )
   | |_____^ expected struct `String`, found struct `password`
```
I'm not looking for a workaround, just thought I'd let you know this was an issue.

--- a/sqlx-macros/src/query/mod.rs
+++ b/sqlx-macros/src/query/mod.rs
@@ -314,6 +314,7 @@ where
                     |&output::RustColumn {
                          ref ident,
                          ref type_,
+                         ..
                      }| quote!(#ident: #type_,),
                 );
 

--- a/sqlx-macros/src/query/output.rs
+++ b/sqlx-macros/src/query/output.rs
@@ -14,6 +14,7 @@ use syn::Token;
 
 pub struct RustColumn {
     pub(super) ident: Ident,
+    pub(super) var_name: Ident,
     pub(super) type_: ColumnType,
 }
 
@@ -114,6 +115,9 @@ fn column_to_rust<DB: DatabaseExt>(describe: &Describe<DB>, i: usize) -> crate::
     };
 
     Ok(RustColumn {
+        // prefix the variable name we use in `quote_query_as!()` so it doesn't conflict
+        // https://github.com/launchbadge/sqlx/issues/1322
+        var_name: quote::format_ident!("sqlx_query_as_{}", decl.ident),
         ident: decl.ident,
         type_,
     })
@@ -129,7 +133,7 @@ pub fn quote_query_as<DB: DatabaseExt>(
         |(
             i,
             &RustColumn {
-                ref ident,
+                ref var_name,
                 ref type_,
                 ..
             },
@@ -140,20 +144,21 @@ pub fn quote_query_as<DB: DatabaseExt>(
                     // binding to a `let` avoids confusing errors about
                     // "try expression alternatives have incompatible types"
                     // it doesn't seem to hurt inference in the other branches
-                    let #ident = row.try_get_unchecked::<#type_, _>(#i)?;
+                    let #var_name = row.try_get_unchecked::<#type_, _>(#i)?;
                 },
                 // type was overridden to be a wildcard so we fallback to the runtime check
-                (true, ColumnType::Wildcard) => quote! ( let #ident = row.try_get(#i)?; ),
+                (true, ColumnType::Wildcard) => quote! ( let #var_name = row.try_get(#i)?; ),
                 (true, ColumnType::OptWildcard) => {
-                    quote! ( let #ident = row.try_get::<::std::option::Option<_>, _>(#i)?; )
+                    quote! ( let #var_name = row.try_get::<::std::option::Option<_>, _>(#i)?; )
                 }
                 // macro is the `_unchecked!()` variant so this will die in decoding if it's wrong
-                (false, _) => quote!( let #ident = row.try_get_unchecked(#i)?; ),
+                (false, _) => quote!( let #var_name = row.try_get_unchecked(#i)?; ),
             }
         },
     );
 
     let ident = columns.iter().map(|col| &col.ident);
+    let var_name = columns.iter().map(|col| &col.var_name);
 
     let db_path = DB::db_path();
     let row_path = DB::row_path();
@@ -165,7 +170,7 @@ pub fn quote_query_as<DB: DatabaseExt>(
 
             #(#instantiations)*
 
-            Ok(#out_ty { #(#ident: #ident),* })
+            Ok(#out_ty { #(#ident: #var_name),* })
         })
     }
 }

